apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ValueConversion.cs

```diff
@@ -328,22 +313,25 @@ private static object ConvertToClrTypeImpl(Value wireValue, V1.Type spannerType,
                     default:
                         throw new ArgumentOutOfRangeException();
                 }
-            if (targetClrType == typeof(IList))
+            if (targetClrType == typeof(ArrayList) || targetClrType == typeof(IList))
+            {
                 switch (wireValue.KindCase)
                 {
-                    case Value.KindOneofCase.NullValue:
-                        return null;
+                    case Value.KindOneofCase.NullValue: return null;
                     case Value.KindOneofCase.ListValue:
-                        return wireValue.ListValue.Values.Select(x => x.ConvertToClrType(spannerType.ArrayElementType))
-                            .ToList();
+                        return new ArrayList(wireValue.ListValue.Values
```

Comment:
I grouped this with ArrayList, so I need to return ArrayList.  Unfortunately there isn't much value in doing a generic collection here -- if the user wants List<T>, then they can pass that as the targetclrtype and it should work (further below).